echo-telemetry/src/main/java/com/netflix/spinnaker/echo/telemetry/TelemetryEventListener.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2019 Armory, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.telemetry;
+
+import com.google.protobuf.util.JsonFormat;
+import com.netflix.spinnaker.echo.events.EchoEventListener;
+import com.netflix.spinnaker.echo.model.Event;
+import com.netflix.spinnaker.proto.stats.Application;
+import com.netflix.spinnaker.proto.stats.Execution;
+import com.netflix.spinnaker.proto.stats.SpinnakerInstance;
+import com.netflix.spinnaker.proto.stats.Stage;
+import com.netflix.spinnaker.proto.stats.Status;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@ConditionalOnProperty("telemetry.enabled")
+public class TelemetryEventListener implements EchoEventListener {
+  private final TelemetryService telemetryService;
+
+  @Autowired
+  public TelemetryEventListener(TelemetryService telemetryService) {
+    this.telemetryService = telemetryService;
+  }
+
+  @Value("${telemetry.instanceId}")
+  String instanceId;
+
+  @Override
+  public void processEvent(Event event) {
+    try {
+      if (event.getDetails() == null
+          || !event.getDetails().getType().equals("orca:pipeline:complete")) {
```

Comment:
Do failed pipelines also send `orca:pipeline:complete`?  I think they might send `orca:pipeline:failed`? If that's the case, none of the status information will be useful as we'll only be recording successful executions.

Reply:
When we are deriving audit log entries from these events, we look for `orca:pipeline:failed`: https://github.com/GoogleCloudPlatform/spinnaker-for-gcp/blob/master/scripts/install/spinnakerAuditLog/index_js.template#L74

Reply:
Good catch. Turns out orchestrations (ad hoc tasks) are also different - added the missing 3